scripts/build/builders/nxp.py

```diff
@@ -330,52 +367,65 @@ def GnBuildArgs(self):
-            if self.os_env == NxpOsUsed.FREERTOS and self.board == NxpBoard.RW61X:
-                flags.append("-DCONFIG_BOOTLOADER_MCUBOOT=true")
-
-        if self.disable_ble:
-            flags.append("-DCONFIG_BT=false")
-
-        if self.enable_wifi:
-            flags.append('-DCONFIG_CHIP_WIFI=true')
-
-        if self.enable_thread:
-            flags.append("-DCONFIG_NET_L2_OPENTHREAD=true -DCONFIG_CHIP_IPV4=false")
-
         if self.enable_factory_data_build:
             # Generate the factory data binary
-            flags.append("-DCONFIG_CHIP_FACTORY_DATA_BUILD=true -DCONFIG_CHIP_FACTORY_DATA=true")
-
-        if self.enable_shell:
-            flags.append("-DCONFIG_CHIP_LIB_SHELL=true")
+            flags.append("-DCONFIG_CHIP_FACTORY_DATA_BUILD=y")
 
         if self.disable_pairing_autostart:
-            flags.append('-DCONFIG_CHIP_ENABLE_PAIRING_AUTOSTART=false')
+            flags.append('-DCONFIG_CHIP_ENABLE_PAIRING_AUTOSTART=n')
 
-        if self.board_variant:
-            flag_board_variant = "-DCONFIG_BOARD_VARIANT=\"%s\"" % self.board_variant.BoardVariantName(self.board)
-            flags.append(flag_board_variant)
+        if self.iw416_transceiver:
+            flags.append('-DCONFIG_MCUX_COMPONENT_component.wifi_bt_module.IW416=y')
+
+        if self.w8801_transceiver:
+            flags.append('-DCONFIG_MCUX_COMPONENT_component.wifi_bt_module.88W8801=y')
+
+        if self.iwx12_transceiver:
+            flags.append('-DCONFIG_MCUX_COMPONENT_component.wifi_bt_module.IW61X=y')
+
+        if self.board == NxpBoard.RT1170:
+            flags.append('-Dcore_id=cm7')
 
         build_flags = " ".join(flags)
 
         return build_flags
 
+    def get_conf_file(self):
+        if self.enable_wifi and self.enable_thread and self.enable_ota and self.enable_factory_data:
+            return "prj_thread_ftd_wifi_br_ota_fdata.conf"
+        elif self.enable_wifi and self.enable_thread and self.enable_ota:
+            return "prj_thread_ftd_wifi_br_ota.conf"
+        elif self.enable_wifi and self.enable_ota and self.enable_factory_data:
+            return "prj_wifi_ota_fdata.conf"
+        elif self.enable_wifi and self.enable_ota:
+            return "prj_wifi_ota.conf"
+        elif self.enable_wifi and self.enable_factory_data:
+            return "prj_wifi_fdata.conf"
+        elif self.enable_wifi and self.disable_ble:
+            return "prj_wifi_onnetwork.conf"
+        elif self.enable_wifi:
+            return "prj_wifi.conf"
+        elif self.enable_thread and self.enable_ota and self.enable_factory_data:
+            return "prj_thread_ftd_ota_fdata.conf"
+        elif self.enable_thread and self.enable_ota:
+            return "prj_thread_ftd_ota.conf"
+        elif self.enable_thread and self.enable_factory_data:
+            return "prj_thread_ftd_fdata.conf"
+        elif self.enable_thread and self.enable_mtd:
+            return "prj_thread_mtd.conf"
+        elif self.enable_thread:
```

Comment:
elif chains should end in a `raise` if nothing matches, since otherwise we will return None and fail with hard to debug failures